eZ/Publish/Core/FieldType/Image/Value.php

```diff
@@ -177,4 +179,27 @@ public function __set($propertyName, $propertyValue)
 
         throw new PropertyNotFoundException($propertyName, get_class($this));
     }
+
+    public function isEquals(SPIValue $value): bool
+    {
+        return $this->toHash($value) === $this->toHash($this);
+    }
+
+    /**
+     * @return array
+     */
+    public function toHash(SPIValue $value)
+    {
+        return [
+            'id' => $value->id,
+            'path' => $value->inputUri ?: $value->id,
+            'alternativeText' => $value->alternativeText,
+            'fileName' => $value->fileName,
+            'fileSize' => $value->fileSize,
+            'uri' => $value->uri,
+            'inputUri' => $value->inputUri,
+            'width' => $value->width,
+            'height' => $value->height,
+        ];
+    }
```

Comment:
Hmm, I see we have a bit of an architecture issue. This is basically a repetition of `\eZ\Publish\Core\FieldType\Image\Type::toHash`, so we cannot have it like that.
Given that details of equality of Value are coded in Type (sic!) we should approach it differently.
```php
class Image\Type extends FieldType implements Comparable
{
    public function valuesEqual(SPIValue $valueA, SPIValue $valueB): bool
}
```
Since `Type`s already contain business logic of hashing, I don't see big issue with adding more logic there.
Any other ideas? POV ping @ViniTou @adamwojs

Reply:
That is kinda what I was suggesting at first here: https://github.com/ezsystems/ezpublish-kernel/pull/3060#pullrequestreview-483530342

So I still agree with it, unless it is a time to chnage `Type::toHash` to facade to `Value::toHash`.